### PR TITLE
Remove the "scary" parts of ConfigPersistence

### PR DESCRIPTION
**Pull request by cgardens**
## What
* Goal: remove `ConfigPersistence` entirely
* This PR removes the "scary" methods from `ConfigPersistence`. These are the ones that dump all records, replace entire tables, and things like that. They are scary because they move a lot of data and have been a headache from the pov of secrets handling. They are also the main things that make `ConfigPersistence` really confusing. After these are gone, it is just a KV interface. We will still want to remove it entirely, but this makes it more innocuous and so it is easier to hide `ConfigPersistence`.

## Recommended reading order
1. easiest to go commit by commit

### CODE REVIEW

File: airbyte-workers/src/main/java/io/airbyte/workers/config/DatabaseBeanFactory.java [hunk 1]
```diff
@@ -92,7 +92,7 @@ public Flyway jobsFlyway(@Named("jobs") final FlywayConfigurationProperties jobs
   @Requires(env = WorkerMode.CONTROL_PLANE)
   public ConfigPersistence configPersistence(@Named("configDatabase") final Database configDatabase,
                                              final JsonSecretsProcessor jsonSecretsProcessor) {
-    return DatabaseConfigPersistence.createWithValidation(configDatabase, jsonSecretsProcessor);
```
Comment (jdpgrailsdev) [inline airbyte-workers/src/main/java/io/airbyte/workers/config/DatabaseBeanFactory.java:95]: We may also be able to remove the creation of the `JsonSecretsProcessor` bean from the `ApplicationBeanFactory` if in fact it is no longer used by any other beans as a result of this change.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit: with one additional comment on some potential additional cleanup.

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryReader.java [hunk 2]
```diff
@@ -87,6 +74,7 @@ private DestinationConnection hydrateDestinationPartialConfig(final DestinationC
     return Jsons.clone(sourceWithPartialConfig).withConfiguration(hydratedConfig);
   }
 
+  @SuppressWarnings("unused")
```
Comment (benmoriceau) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryReader.java:77]: Should we remove it if this is unused?

File: airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryWriter.java [hunk 3]
```diff
@@ -123,6 +115,7 @@ public void writeDestinationConnection(final DestinationConnection destination,
    * @param spec connector specification
    * @return partial config
    */
+  @SuppressWarnings("unused")
```
Comment (benmoriceau) [inline airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/SecretsRepositoryWriter.java:118]: Same comment about the unused

**Review by benmoriceau — APPROVED** [review]

**Pull request merged**